nengo/utils/tests/test_socket.py

```diff
@@ -0,0 +1,151 @@
+import math
+import threading
+import numpy as np
+
+import nengo
+from nengo.utils import sockets
+
+
+class SimThread(threading.Thread):
+    def __init__(self, simulator, sim_time, lock=None, wait=False):
+        threading.Thread.__init__(self)
+        self.sim = simulator
+        self.sim_time = sim_time
+        self.lock = lock
+        self.wait = wait
+
+    def run(self):
+        if self.lock is not None and self.wait:
```

Comment:
Why would `self.lock == None` anyways? If that's something you want to check for, shouldn't you check for it in `__init__()`?

Reply:
possibly the lock might be handling some weird race conditions on startup? if i'm reading it right it's only ever catching here once, then the lock is set and never cleared